src/app/Nonagent/main.py

```diff
@@ -0,0 +1,594 @@
+    # statuses -	List of specific statuses to report on.  Leave blank for all, or use dash delimiters if
+    # requesting more than one status.
+    # categories -	List of specific status categories to report on.  Leave blank for all, or use dash delimiters
+    # if requesting more than one category.
+    # users -		List of specific users to report on.  Use single dash delimiters if requesting more than
+    # one specific user
+    # status_breakdown -	[0,1] Breakdown of all statuses within selected elements, default 0
+    # show_percentages -	[0,1] (Only works if status_breakdown above is enabled), will show percentages of statuses,
+    # default 0
+    # file_download -		[0,1] Download as a CSV file, default 0
+
+    'update_campaign',
+    # updates campaign information in the vicidial_campaigns table
+    # For required field check update_campaign field in documentation: http://vicidial.org/docs/NON-AGENT_API.txt
+
+    'update_alt_url',
+    # updates/adds/displays alternate dispo call url entries for a campaign
+    # For required field check update_alt_url field in documentation: http://vicidial.org/docs/NON-AGENT_API.txt
+
+    'update_presets',
+    # updates/adds/displays/deletes campaign preset entries
+    #
+    # NOTE: api user for this function must have user_level set to 8 or higher and "modify campaigns" enabled
+    #
+    # REQUIRED FIELDS-
+    # campaign_id -		2-8 characters
+    # source -		description of what originated the API call (maximum 20 characters)
+    # preset_name -		name of the preset in the system
+    # 			   NOTE: If only one preset exists for this campaign, then this field can be left blank
+    # action -		'UPDATE', 'NEW', 'DELETE' or 'LIST' (default is 'UPDATE')
+    # 			   NOTE: 'stage'(csv, tab, pipe[default]) and 'header'(YES, NO) options are available for an action
+    # 			   of 'LIST'
+    #
+    # EDITABLE FIELDS-
+    # preset_hide_number -	One of these: 'Y','N', default is 'N'
+    # preset_number -		Digits only (must be from 1 to 50 digits in length)
+    # preset_dtmf -		Digits and certain characters only
+    # ('0123456789PSQ' [P = #-pound, S = *-star, Q = half-second-quiet])
+
+    'add_did',
+    # adds new Inbound DID entries to the system in the vicidial_inbound_dids table
+    # For required field check add_did field in documentation: http://vicidial.org/docs/NON-AGENT_API.txt
+
+    'update_did',
+    # updates Inbound DID information in the vicidial_inbound_dids table
+    # For required field check update_did field in documentation: http://vicidial.org/docs/NON-AGENT_API.txt
+
+    'update_cid_group_entry',
+    # updates CID Group entries in the vicidial_campaign_cid_areacodes table
+    # For required field check update_cid_group_entry field in documentation: http://vicidial.org/docs/NON-AGENT_API.txt
+]
+
+
+async def call_non_agent_api(base_non_agent_url: str, function_and_values_dict: dict) -> dict:
```

Comment:
isn't it better to define each query param instead of having a dictionary? this way you don't need to check presence of _function_ in query params

Reply:
I'm not familiar with the api but if there are some permanent query params then better to separate them from the dict

Reply:
<img width="1235" alt="Screenshot 2022-11-09 at 09 35 25" src="https://user-images.githubusercontent.com/87501969/200780362-161b0893-caf8-439e-b9af-ffc227fc9efa.png">
This is an example, only user, ip, pass and source is in almost every function and these ones are part of base url, rest of them are changing.